src/scrapers/isracard.ts

```diff
@@ -0,0 +1,217 @@
+            });
+            txnIsrael.push(...txns);
+          }
+
+          if (txnGroup.txnAbroad) {
+            const txns = txnGroup.txnAbroad.map((txn: any) => {
+              return Object.assign(txn, {
+                processedDate: account.processedDate,
+              });
+            });
+            txnAbroad.push(...txns);
+          }
+        });
+        const accountNum = lodash.get(account, 'accountNumber');
+        accountTxns[accountNum] = {
+          accountNumber: accountNum,
+          index: account.index,
+          txnIsrael: txnIsrael,
+          txnAbroad: txnAbroad,
+        };
+      }
+    });
+    return accountTxns;
+  }
+  return 0;
+}
+
+async function getMonthDashboard(
+  page: puppeteer.Page,
+  monthMoment: moment.Moment,
+  options?: isracardOptions
+) {
+  // get accounts data
+  const billingDate = monthMoment.format('YYYY-MM-DD');
+  const accountsUrl = `${SERVICE_URL}?reqName=DashboardMonth&actionCode=0&billingDate=${billingDate}&format=Json`;
+  const data = fetchGetWithinPage<IsracardDashboardMonth>(page, accountsUrl);
+
+  if (options && options.validateSchema) {
+    await data;
+    let validation = await validateSchema(isracardDashboardMonth, data);
+    Object.assign(data, validation);
+  } else {
+    return data;
+  }
+}
+
+async function getMonthTransactions(
+  page: puppeteer.Page,
+  monthMoment: moment.Moment,
+  options?: isracardOptions
+) {
+  /* get transactions data */
+  const month = monthMoment.month() + 1;
+  const monthStr = month < 10 ? `0${month}` : month.toString();
+  const transUrl = `${SERVICE_URL}?reqName=CardsTransactionsList&month=${monthStr}&year=${monthMoment.year()}&requiredDate=N`;
+  const data = fetchGetWithinPage<IsracardCardsTransactionsList>(
+    page,
+    transUrl
+  );
+
+  if (options && options.validateSchema) {
+    await data;
+    let validation = await validateSchema(isracardCardsTransactionsList, data);
+    Object.assign(data, validation);
+    return data;
+  } else {
+    return data;
+  }
+}
+
+export async function isracard(
+  page: puppeteer.Page,
+  options?: isracardOptions
+) {
+  const BASE_URL = 'https://digital.isracard.co.il';
+  await page.goto(`${BASE_URL}/personalarea/Login`);
+
+  await login(page);
+
+  /* dates logic  */
+  let startMoment = moment().subtract(5, 'years').startOf('month');
+  const allMonths: moment.Moment[] = [];
+  let lastMonth = moment().startOf('month').add(1, 'month');
+  while (startMoment.isSameOrBefore(lastMonth)) {
+    allMonths.push(startMoment);
+    startMoment = moment(startMoment).add(1, 'month');
+  }
+
+  return {
+    getDashboard: async () => {
+      return Promise.all(
+        /* get monthly results */
+        allMonths.map(async (monthMoment) => {
+          return getMonthDashboard(page, monthMoment, options);
+        })
+      );
+    },
+    getTransactions: async () => {
+      return Promise.all(
+        /* get monthly results */
+        allMonths.map(async (monthMoment) => {
```

Comment:
where does monthMoment comes from?
can we have a default behavior that goes back as long in the past as possible?

Reply:
I'm not sure how long is possible.
3 years? since the a specific date?

I get data since Jan. 2018. why since then?

if I knew it, I'd code it to get all possible...

Reply:
I've exposed the `getMonthTransactions` so I care a bit less about it now, but if I would use those, I think maybe the default 2 and a half?
Until today I always used 2 years back, but maybe 2 and a half is the right one?
Can you try it out on your account?
Maybe 2 and 6 months +- 1 month check

Reply:
added corresponding 'options' attribute.
default set to 30 months (2.5 years)